app/components-react/windows/ManageSceneCollections.tsx

```diff
@@ -0,0 +1,191 @@
+  }
+
+  function goToThemes() {
+    NavigationService.actions.navigate('BrowseOverlays');
+    WindowsService.actions.closeChildWindow();
+  }
+
+  return (
+    <ModalLayout onCancel={close}>
+      <Layout style={{ height: '100%' }}>
+        <Sider width={300}>
+          <div>{$t('Your Scene Collections:')}</div>
+          <div style={{ width: '96%' }}>
+            <TextInput
+              placeholder={$t('Search Scene Collections')}
+              onChange={setQuery}
+              uncontrolled={false}
+              addonBefore={<i className="icon-search" />}
+              nowrap
+            />
+          </div>
+          <Scrollable style={{ height: 'calc(100% - 32px)' }}>
+            {filteredCollections().map((collection, i) => (
+              <CollectionNode collection={collection} recentlyUpdated={i < 2} key={collection.id} />
+            ))}
+          </Scrollable>
+        </Sider>
+        <Content style={{ paddingLeft: '24px' }}>
+          <div>{$t('Add New Scene Collection:')}</div>
+          <div className={styles.buttonContainer}>
+            <button onClick={create} className={cx('button', styles.button)}>
+              <i className="icon-stream-labels" />
+              <strong>{$t('New')}</strong>
+              <p>{$t('Start fresh and build from scratch')}</p>
+            </button>
+            <button onClick={importFromObs} className={cx('button', styles.button)}>
+              <i className="icon-cloud-backup" />
+              <strong>{$t('Import')}</strong>
+              <p>{$t('Load existing scenes from OBS')}</p>
+            </button>
+            <button onClick={goToThemes} className={cx('button', styles.button, styles.lg)}>
+              <div>
+                <strong>{$t('Template')}</strong>
+                <p>{$t('Choose a template from our theme library')}</p>
+              </div>
+              <img src={$i('images/prime-themes.png')} />
+            </button>
+          </div>
+        </Content>
+      </Layout>
+    </ModalLayout>
+  );
+}
+
+function CollectionNode(p: {
+  collection: ISceneCollectionsManifestEntry;
+  recentlyUpdated: boolean;
+}) {
+  const { SceneCollectionsService } = Services;
+  const [duplicating, setDuplicating] = useState(false);
+  const modified = moment(p.collection.modified).fromNow();
+  const isActive = p.collection.id === SceneCollectionsService.activeCollection?.id;
+
+  useEffect(onNeedsRenamedChanged, [p.collection.needsRename]);
+
+  function onNeedsRenamedChanged() {
+    if (p.collection.needsRename) rename();
+  }
+
+  function makeActive() {
+    if (p.collection.operatingSystem !== getOS()) return;
+    SceneCollectionsService.actions.load(p.collection.id);
+  }
+
+  function duplicate() {
+    setDuplicating(true);
+
+    setTimeout(() => {
+      SceneCollectionsService.actions.return
+        .duplicate(p.collection.name, p.collection.id)
+        .then(() => setDuplicating(false))
+        .catch(() => setDuplicating(false));
```

Comment:
FYI promises these days have a function called `finally()` that lets you schedule a callback for when the promise is settled, regardless or whether it resolves or rejects.  It's useful for behavior that you want to ensure always happens when the operation finishes, regardless of result.